### ISSUE TITLE
Support switching the game language at runtime

### ISSUE THREAD

**Issue by chungy** [RobertBeckebans/RBDOOM-3-BFG#389]
The GOG.com release of BFG includes all language files out of the base install and added a Language.ini file that can change it between English/German/Japanese/etc. AFAIK, code doesn't exist in the public source (nor this fork) for switching between them.

**Comment by shmerl** [comment]
Yeah, I was bitten by this, and by default it set it as Japanese for me. I had to add this to defaults.cfg:

```seta sys_lang "english"```

**Comment by BielBdeLuna** [comment]
What does the Language.ini contain? and where is that file?

**Comment by shmerl** [comment]
@BielBdeLuna: It's not part of the game, I think it's GOG's installer related. Unpacking it with innoextract, I see these files:

    ./app/__support/app/Language.ini

```
[Game]
Language=English
```

    ./app/language_setup.ini

```
[Info]
GameID=1135892318
[Options]
Option1=English
Option2=French
Option3=German
Option4=Italian
Option5=Spanish
Option6=Japanese

[English]
IniPath={app}/language.ini
Section=Game
Key=Language
Value=English
[French]
IniPath={app}/language.ini
Section=Game
Key=Language
Value=French
[German]
IniPath={app}/language.ini
Section=Game
Key=Language
Value=German
[Italian]
IniPath={app}/language.ini
Section=Game
Key=Language
Value=Italian
[Spanish]
IniPath={app}/language.ini
Section=Game
Key=Language
Value=Spanish
[Japanese]
IniPath={app}/language.ini
Section=Game
Key=Language
Value=Japanese
```

**Comment by shmerl** [comment]
Anyway, the proper way to set the language is in `default.cfg` in the base directory, or in `D3BFGConfig.cfg` in your local one.

**Comment by BielBdeLuna** [comment]
yes, and by the looks of it, adding this wouldn't be the same as adding functionality, so yeah, in either `default.cfg` or in `D3BFGConfig.cfg` or via console command `seta sys_lang "english"` keeps being the right way.

**Comment by BielBdeLuna** [comment]
maybe we should prioritize English over Japanese

**Comment by shmerl** [comment]
Yeah, I suppose it picks Japanese by default.

**Comment by PaddyMac** [comment]
I ran into this issue while running it under Linux. Adding set sys_lang "english" to ~/.rbdoom3bfg/base/D3BFGConfig.cfg does fix the issue, but the problem is that every time I shut down the game, it removes that entry from the config file. So I must add it again every time I run the game. Either that, or I need to learn Japanese. I tried chmod 444 on the config file, and the game started without any apparent problem, but it caused it to segfault on exit.

**Comment by garoto** [comment]
What about adding that line to `autoexec.cfg`?

**Comment by PaddyMac** [comment]
Adding that line to autoexec.cfg did the trick! Thank you for suggesting it. By the way, the chmod thing apparently had nothing to do with the game segfaulting on exit. It does that on exit anyway for some reason. That basically just results in me having to manually reset my desktop resolution after exiting. Its a minor nuisance, but that's another issue.
